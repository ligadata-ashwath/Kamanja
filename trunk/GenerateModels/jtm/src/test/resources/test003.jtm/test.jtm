{

  "header": {
    "namespace": "com.ligadata.jtm.test.filter",
    "version": "0.0.1",
    "language" : "scala",
    "minVersion" : "2.11"
  },

  "dependencies": ["list of jars/modules - this should go into command line arguments instead of this file"],

  "imports": {
    "packages": [ "com.ligadata.Utils._" ]
  },

  "aliases": {
    "messages": {
      "m1": "com.ligadata.kamanja.test.msg1",
      "m2": "com.ligadata.kamanja.test.msg3",
      "omsg2": "com.ligadata.kamanja.test.msg2",
      "omsg3": "com.ligadata.kamanja.test.msg4",
      "c1": "com.ligadata.kamanja.test.msg5",
      "c2": "com.ligadata.kamanja.test.msg6"
    }
  },

  "transformations": {
    "test1": {
      "dependsOn": [ ["m1", "m2", "c1"], ["m2", "c1", "c2"]],
      "computes": {
        "zipcode": { "type": "Int", "val": "iplookup(${m2.in1}).zipcode", "comment": "in scala, type could be optional" },
        "zipcode1": { "type": "Int", "vals": ["iplookup(${m1.in1}).zipcode", "iplookup(${m2.in1}.zipcode"] },
        "ipinfo": { "type": "IpInfo", "val": "iplookup(${m2.in1})",  "comment": "type could be some class available from imports" }
      },

      "comment": "msg2 is part of com.ligadata.jtm.test.filter",

      "outputs": {
        "omsg2": {
          "mapping": {
            "out1": "${m2.in1}",
            "out2": "tmp8",
            "out3": "somefunc($zipcode, $zipcode1)",
            "out4": "$ipinfo.zipCode"
          },
          "where": "!(${m2.in2} != -1 && ${m2.in2} < 100)",
          "computes": {
            "tmp8": { "type": "String", "val":"somefunc2(${m2.in3})" }
          }
        }
      }
    },
    "test2": {
      "dependsOn": [ ["com.ligadata.kamanja.test.msg3"] ],
      "outputs": {
        "omsg3": {
          "mappingByName": true,
          "mapping": {
            "out1": "in1",
            "out2": "in1",
            "out3": "in1",
            "out4": "in1"
          }
        }
      }
    }
  }
}
