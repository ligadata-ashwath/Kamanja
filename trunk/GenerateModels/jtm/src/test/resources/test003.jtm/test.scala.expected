/*
* Copyright 2016 ligaDATA
*
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
*
*     http://www.apache.org/licenses/LICENSE-2.0
*
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
*/
package com.ligadata.jtm.test.filter.V1
import com.ligadata.KamanjaBase._
import com.ligadata.KvBase.TimeRange
import com.ligadata.kamanja.metadata.ModelDef
import com.ligadata.runtime.Log
import com.ligadata.Utils._
import com.ligadata.runtime.Conversion
class ModelFactory(modelDef: ModelDef, nodeContext: NodeContext) extends ModelInstanceFactory(modelDef, nodeContext) {
  override def createModelInstance(txnCtxt: com.ligadata.KamanjaBase.TransactionContext): ModelInstance = return new Model(this)
  override def getModelName: String = "com.ligadata.jtm.test.filter.Model"
  override def getVersion: String = "0.0.1"
  override def createResultObject(): ModelResultBase = new MappedModelResults()
}
class Model(factory: ModelInstanceFactory) extends ModelInstance(factory) {
  val conversion = new com.ligadata.runtime.Conversion
  val log = new com.ligadata.runtime.Log(this.getClass.getName)
  val context = new com.ligadata.runtime.JtmContext
  import log._
  // Model code start
  // Model code end
  override def execute(txnCtxt: TransactionContext, execMsgsSet: Array[ContainerOrConcept], triggerdSetIndex: Int, outputDefault: Boolean): Array[ContainerOrConcept] = {
    if (isTraceEnabled)
    Trace(s"Model::execute transid=%d triggeredset=%d outputdefault=%s".format(txnCtxt.transId, triggerdSetIndex, outputDefault.toString))
    if(isDebugEnabled)
    {
      execMsgsSet.foreach(m => Debug( s"Input: %s -> %s".format(m.getFullTypeName, m.toString())))
    }
    // Grok parts
    // Model methods
    // msg2 is part of com.ligadata.jtm.test.filter
    def exeGenerated_test1_1(msg1: com.ligadata.kamanja.test.V1000000.msg1, msg2: com.ligadata.kamanja.test.V1000000.msg3, msg3: com.ligadata.kamanja.test.V1000000.msg5): Array[MessageInterface] = {
      Debug("exeGenerated_test1_1")
      context.SetSection("test1")
      // in scala, type could be optional
      val zipcode: Int = iplookup(msg2.in1).zipcode
      val zipcode1: Int = iplookup(msg1.in1).zipcode
      // type could be some class available from imports
      val ipinfo: IpInfo = iplookup(msg2.in1)
      def process_omsg2(): Array[MessageInterface] = {
        Debug("exeGenerated_test1_1::process_omsg2")
        context.SetScope("omsg2")
        try {
          if (!(!(msg2.in2 != -1 && msg2.in2 < 100))) {
            Debug("Filtered: test1@omsg2")
            return Array.empty[MessageInterface]
          }
          val tmp8: String = somefunc2(msg2.in3)
          val result = com.ligadata.kamanja.test.V1000000.msg2.createInstance
          result.out4 = ipinfo.zipCode
          result.out3 = somefunc(zipcode, zipcode1)
          result.out2 = tmp8
          result.out1 = msg2.in1
          if(context.CurrentErrors()==0) {
            Array(result)
          } else {
            return Array.empty[MessageInterface]
          }
        } catch {
          case e: Exception => {
            context.AddError(e.getMessage)
          }
        }
        return Array.empty[MessageInterface]
      }
      process_omsg2()
    }
    // msg2 is part of com.ligadata.jtm.test.filter
    def exeGenerated_test1_2(msg2: com.ligadata.kamanja.test.V1000000.msg3, msg3: com.ligadata.kamanja.test.V1000000.msg5, msg4: com.ligadata.kamanja.test.V1000000.msg6): Array[MessageInterface] = {
      Debug("exeGenerated_test1_2")
      context.SetSection("test1")
      // in scala, type could be optional
      val zipcode: Int = iplookup(msg2.in1).zipcode
      val zipcode1: Int = iplookup(msg2.in1.zipcode
      // type could be some class available from imports
      val ipinfo: IpInfo = iplookup(msg2.in1)
      def process_omsg2(): Array[MessageInterface] = {
        Debug("exeGenerated_test1_2::process_omsg2")
        context.SetScope("omsg2")
        try {
          if (!(!(msg2.in2 != -1 && msg2.in2 < 100))) {
            Debug("Filtered: test1@omsg2")
            return Array.empty[MessageInterface]
          }
          val tmp8: String = somefunc2(msg2.in3)
          val result = com.ligadata.kamanja.test.V1000000.msg2.createInstance
          result.out4 = ipinfo.zipCode
          result.out3 = somefunc(zipcode, zipcode1)
          result.out2 = tmp8
          result.out1 = msg2.in1
          if(context.CurrentErrors()==0) {
            Array(result)
          } else {
            return Array.empty[MessageInterface]
          }
        } catch {
          case e: Exception => {
            context.AddError(e.getMessage)
          }
        }
        return Array.empty[MessageInterface]
      }
      process_omsg2()
    }
    def exeGenerated_test2_3(msg2: com.ligadata.kamanja.test.V1000000.msg3): Array[MessageInterface] = {
      Debug("exeGenerated_test2_3")
      context.SetSection("test2")
      def process_omsg3(): Array[MessageInterface] = {
        Debug("exeGenerated_test2_3::process_omsg3")
        context.SetScope("omsg3")
        try {
          val result = com.ligadata.kamanja.test.V1000000.msg4.createInstance
          result.out4 = msg2.in1
          result.out3 = msg2.in1
          result.out2 = conversion.ToString(msg2.in1)
          result.out1 = msg2.in1
          if(context.CurrentErrors()==0) {
            Array(result)
          } else {
            return Array.empty[MessageInterface]
          }
        } catch {
          case e: Exception => {
            context.AddError(e.getMessage)
          }
        }
        return Array.empty[MessageInterface]
      }
      process_omsg3()
    }
    // Evaluate messages
    val msgs = execMsgsSet.map(m => m.getFullTypeName -> m).toMap
    val msg1 = msgs.getOrElse("com.ligadata.kamanja.test.msg1", null).asInstanceOf[com.ligadata.kamanja.test.V1000000.msg1]
    val msg2 = msgs.getOrElse("com.ligadata.kamanja.test.msg3", null).asInstanceOf[com.ligadata.kamanja.test.V1000000.msg3]
    val msg3 = msgs.getOrElse("com.ligadata.kamanja.test.msg5", null).asInstanceOf[com.ligadata.kamanja.test.V1000000.msg5]
    val msg4 = msgs.getOrElse("com.ligadata.kamanja.test.msg6", null).asInstanceOf[com.ligadata.kamanja.test.V1000000.msg6]
    // Main dependency -> execution check
    // Create result object
    val results: Array[MessageInterface] =
    (if(msg1!=null && msg2!=null && msg3!=null) {
      exeGenerated_test1_1(msg1, msg2, msg3)
    } else {
      Array.empty[MessageInterface]
    }) ++
    (if(msg2!=null && msg3!=null && msg4!=null) {
      exeGenerated_test1_2(msg2, msg3, msg4)
    } else {
      Array.empty[MessageInterface]
    }) ++
    (if(msg2!=null) {
      exeGenerated_test2_3(msg2)
    } else {
      Array.empty[MessageInterface]
    }) ++
    Array.empty[MessageInterface]
    if(isDebugEnabled)
    {
      results.foreach(m => Debug( s"Output: %s -> %s".format(m.getFullTypeName, m.toString())))
    }
    results.asInstanceOf[Array[ContainerOrConcept]]
  }
}
